hi-ml-histopathology/testhisto/testhisto/datasets/test_panda_slides.py

```diff
@@ -0,0 +1,91 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+from pathlib import Path
+from typing import Any, List
+from unittest import mock
+from unittest.mock import MagicMock
+
+import numpy as np
+import pytest
+import torch
+from pytorch_lightning import seed_everything
+
+from histopathology.configs.classification.DeepSMILESlidesPandaBenchmark import DeepSMILESlidesPandaBenchmark
+from histopathology.utils.naming import SlideKey
+from testhisto.mocks.base_data_generator import MockHistoDataType
+from testhisto.mocks.slides_generator import MockPandaSlidesGenerator, TilesPositioningType
+
+try:
+    from cucim import CuImage  # noqa: F401
+    has_cucim = True
+except:  # noqa: E722
+    has_cucim = False
+
+
+@pytest.mark.skipif(not has_cucim, reason="Test requires CUCIM library")
+@pytest.mark.gpu
+def test_panda_reproducibility(tmp_path: Path) -> None:
```

Comment:
Is it somehow possible to write a DDP test as well? We might want to make sure that everything is properly seeded in DDP context.  You can use `run_distributed` from utils_testhisto.